### ISSUE TITLE
OCRmyPDF can now be installed in an easier way on FreeBSD

### ISSUE THREAD

**Issue by knobix** [jbarlow83/OCRmyPDF#417]
**Describe the issue**

OCRmyPDF is available as a Package on FreeBSD (all releases) since July 2019 (see [here](https://www.freshports.org/textproc/py-ocrmypdf/) for further details). 

Users don't need to build pikepdf and install the other required components manually anymore. Instead they can install OCRmyPDF with all required dependencies simply by issuing:

`# pkg install py36-ocrmypdf
`

**Expected behavior**

The parts in the documentation regarding FreeBSD can be updated on occasion.

**System:**

- OS: FreeBSD (11.2-, 11.3-, 12.0-RELEASE and 13.0-CURRENT)
- OCRmyPDF Version: 8.3.2

**Additional context**

At the moment I'm working to update OCRmyPDF to the 9.0.0 release but I've to update the outdated port of [graphics/unpaper](https://www.fres hports.org/graphics/unpaper/) first before I can proceed. As far I can see some tests of the 9.0.0 release requires "unpaper" unconditionally to get a successful result.

At last but not least: Thank you for creating OCRmyPDF and keep up the good work!

**Comment by jbarlow83** [comment]
Thank you!

I have a commit to address the tests that fail. pngquant, jbig2enc and unpaper are all optional, but should be provided where possible. I also added a minimal test to CI to confirm this will keep working. v9.0.1 should fix this and document the availability of your ports. 

I peeked at your Makefile - of course I can't see 9.0.0. One quick note - if FreeBSD handles shell completions, you could also install the files in `misc/completions` to the system directories for bash and fish completions respectively. I haven't seen pytest issue warnings when some tests don't run but if you have any suggestions about changing that workaround for the `slow` tests I'm happy to. There's very little point in anyone downstream running those.

**Comment by knobix** [comment]
Thank you for the quick fixes, hints and the bump to release 9.0.1! I'll update the FreeBSD port to that version soon.

The updated port will then include the shell completions and omits the "slow" tests. Without the `--runslow` parameter the whole test will finish successfully. The workaround with the "slow" marker would be to add following line to `setup.cfg` under the `[pytest]` section:

`markers = slow
`

pngquant will change from a "hard" requirements to a "soft" requirement in the FreeBSD Makefile by creating an option for it and enable that option by default. I'll do so in the same way for unpaper and jbig2enc in near future.

Apropos pngquant: At the moment the test `test_mono_not_inverted` is skipped in the FreeBSD Makefile because it emits an assertion error. I'm not quite sure if #411 is related to this, otherwise I'll open a separate issue for it.

**Comment by jbarlow83** [comment]
setup.cfg already has 

```
[tool:pytest]
norecursedirs = lib .pc .git output cache resources
testpaths = tests
filterwarnings =
	ignore:.*XMLParser.*:DeprecationWarning
markers =
	slow
```

**Comment by knobix** [comment]
Thanks for the pointer! Indeed, since the 9.0.0 release that workaround is no longer required.

**Comment by jbarlow83** [comment]
The assert is not related to #411. That test also does not involve pngquant, it uses Ghostscript and Pillow. If it fails that means black and white will be inverted on the output in some cases. Any more information on that would be helpful.

**Comment by knobix** [comment]
That assert will only occur if pngquant is present in the system which is a bit strange then. The output is:
```
E       AssertionError: Expected white background
E       assert 0 == 255
E         -0
E         +255
```
Some info about the other components and the OS:

- OS: FreeBSD (all releases, amd64 + i386)
- Python: 3.6.9
- Pillow:  6.0.0 (with compiled-in support for LCMS, FreeType, JPEG, OpenJPEG, PNG, TIFF, Tkinter BitmapImage & PhotoImage and webp)
- Ghostscript: 9.27 (without compiled-in support for OpenJPEG)
- pngquant: 2.12.3 (without compiled-in support for LCMS)

Don't hesitate to ask if you need more info.

**Comment by knobix** [comment]
Some additional notes / observations regarding the `test_mono_not_inverted` issue:

The test passes with pngquant present in the system if the `level` parameter of `opt.main()` is set to `0` or `1` instead of `3`.

The following info about qpdf might also be helpful:

- qpdf: 8.4.2

**Comment by jbarlow83** [comment]
Thanks, I was mistaken, the result will vary based on pngquant. 

I suspect it has to do with the omission of LCMS, since Debian, Fedora and macOS all have LCMS.

I suggest excluding pngquant as a dependency and marking it as a conflicting package until that package maintainer can work out LCMS. PDFs are color managed and we shouldn't be passing data through programs that aren't color managed. Or, if there's a way to jail (?) ocrmypdf so that it can't see a pngquant install that would do it too....

**Comment by knobix** [comment]
I did some testing meanwhile and can give some new insights. It might be that pngquant isn't the real reason for it but I haven't found the root cause, yet.

The `test_mono_not_inverted` check even fails with a version of pngquant that has support for LCMS.

I was able to get successful results together with a usable/compressed `out.pdf` if these lines in `src/ocrmypdf/optimize.py` are commented out:
```
if compdata.bps == 1:
    im_obj.Decode = [1, 0]  # Bit of a kludge but this inverts photometric too
```

**Comment by jbarlow83** [comment]
Wait, so the test suite passes when that is commented out, and the output PDF when viewed in a PDF viewer shows black text on a white page?

`pngquant` behaves differently in some cases :\

**Comment by knobix** [comment]
Yes, here's a list of the tests I did and the results. I have the PNG/PDF files for the case if you need some more info about them:

### Tests with pngquant (without LCMS2 support):

Test without `pngquant` present in the system:

- `im.png` -> white with three black dots
- `out.pdf` -> black letters on white background

Test with `pngquant` present in the system:

- `im.png` -> black with three white dots
- `out.pdf` -> white letters on black background

Test with `pngquant` present and commented out lines in `src/ocrmypdf/optimize.py`:

- `im.png` -> white with three black dots
- `out.pdf` -> black letters on white background

Test without `pngquant` present and commented out lines in `src/ocrmypdf/optimize.py`:

- `im.png` -> white with three black dots
- `out.pdf` -> black letters on white background

### Tests with pngquant (with LCMS2 support)

Test without `pngquant` present in the system:

- `im.png` -> white with three black dots
- `out.pdf` -> black letters on white background

Test with `pngquant` present in the system:

- `im.png` -> black with three white dots
- `out.pdf` -> white letters on black background

Test with `pngquant` present and commented out lines in `src/ocrmypdf/optimize.py`:

- `im.png` -> white with three black dots
- `out.pdf` -> black letters on white background

Test without `pngquant` present and commented out lines in `src/ocrmypdf/optimize.py`:

- `im.png` -> white with three black dots
- `out.pdf` -> black letters on white background
